net/dns/direct.go

```diff
@@ -481,10 +497,25 @@ func (m *directManager) checkForFileTrample() {
 	}
 	m.logf("trample: resolv.conf changed from what we expected. did some other program interfere? current contents: %q", show)
 	m.health.SetUnhealthy(resolvTrampleWarnable, nil)
+	if m.trampleDNSPub != nil {
+		m.trampleCount.Add(1)
+		if m.trampleCount.Load() < 10 {
+			m.trampleDNSPub.Publish(TrampleDNS{
+				LastTrample:       time.Now(),
+				TramplesInTimeout: m.trampleCount.Load(),
+			})
+			m.trampleTimer.Reset(trampleWatchDuration)
+		} else {
+			m.logf("trample: resolv.conf was trampled too many times (%d). Abandoning trying to fix it.", m.trampleCount.Load())
+		}
```

Comment:
You could do this without an add-then-load:
```suggestion
		n := m.trampleCount.Add(1)
		if n < 10 {
			m.trampleDNSPub.Publish(TrampleDNS{
				LastTrample:       time.Now(),
				TramplesInTimeout: n,
			})
			m.trampleTimer.Reset(trampleWatchDuration)
		} else {
			m.logf("trample: resolv.conf was trampled too many times (%d). Abandoning trying to fix it.", n)
		}
```